backend/flaskr/users/handlers.py

```diff
@@ -59,18 +69,19 @@ def make_user():
         mk_error("Password must be at least 5 characters long")
     )
 
-    # TODO: Maybe some email format check?
+    email_format = lambda: lambda x: re.match(r'[A-Za-z][A-Za-z0-9_\.]*@[A-Za-z0-9_]+\.[A-Za-z0-9_]')
```

Comment:
No need for curried lamda here.

` min_size = lambda n: lambda x: len(str(x)) >= n` 

First lambda hare was used as a factory for creating predicates. In email_fomat case code below woud work just fine.

email_format = lambda x: re.match(r'[A-Za-z][A-Za-z0-9_\.]*@[A-Za-z0-9_]+\.[A-Za-z0-9_]')
validator.field_predicate(
        "email", email_format,
        mk_error("Incorrect email format")
    )